### PR TITLE
Skip deploying services that do not need to be re-run

### PR DESCRIPTION
**Pull request by mojotalantikite**
## Summary

In our environment Serverless unfortunately will spend a lot of time redeploying things even if there are no changes to the service it is deploying. (Some of this additional time is because of Serverless needing to run webpack.) We thought it'd be nice to skip redeploying if a service has been successfully deployed and not changed in the last commit. This PR begins to do that.

Currently we are only skipping services that have changed since a last successful CI run. Taking into account services that succeeded in a run that ultimately failed starts getting complicated, as a skipped service in the context of a failure doesn't mean the past run succeeded. In our DTBM meeting we decided that this was fine for our first pass.

#### Related issues
https://qmacbis.atlassian.net/browse/OY2-14971

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -155,9 +184,10 @@ jobs:
 
   deploy-infra:
     needs: [begin-deployment]
-    uses: CMSgov/managed-care-review/.github/workflows/deploy-infra-to-env.yml@main
+    uses: CMSgov/managed-care-review/.github/workflows/deploy-infra-to-env.yml@mt-skip-deploys
```
Comment (haworku) [inline .github/workflows/deploy.yml:187]: don't forget to change this back after merge heh

File: scripts/get-changed-services/index.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+// just for testing locally now
+function readToken(path = '../../../../../access_token.txt') {
+    return fs.readFileSync(path).toString().trim()
+}
+
+process.env.GITHUB_ACTION = 'true'
+process.env.GITHUB_TOKEN = readToken()
+*/
+
+// for now keep this list hardcoded
```
Comment (haworku) [inline scripts/get-changed-services/index.ts:17]: urgh. Would you add this section to our README plz

"How to add a new service"
- make a folder in `/services`. 
- add a `package.json`,  `serverless.yml` and a `README.md` inside
- if you plan to write js/ts code, make a a `src` folder first. This is a Node standard.
- to add your services to our deploy flow - Add a step to `deploy-app-to- env` and add your service to list in `get-changed-services` scripts

Comment (mojotalantikite) [reply scripts/get-changed-services/index.ts:7]: Sounds good, I'll add a section like this to the README. 

I can also give getting this list of services programmatically another go, I have a func in this action that pulls it from Lerna, but hardcoded to get myself going faster.

Comment (haworku) [reply scripts/get-changed-services/index.ts:7]: yeah thats fine just want to start documenting stuff like this.

**Review by haworku — APPROVED** [review]
From what I can tell the main way we will test this is over time as we merge PRs.

File: README.md [hunk 3]
```diff
@@ -248,16 +248,29 @@ to Cloudtamer.
...
+-   `lerna create ${service-name}`. Follow Lerna's prompts and you'll end up with a directory under `./services/` with a generated `package.json` and `README.md` file.
+-   Add a `serverless.yml` file to the root directory of this new service. You can copy off of an existing config or run the `serverless` command in `./services/${service-name}` to use one of their starter templates.
+-   If this service is going to require js or ts code, you'll want to create a `src` directory as well as copy over the appropriate `tsconfig.json` and `.eslintrc` configs. Refer to one of the existing services to get an idea of how we are currently doing this.
+
+You'll need to add this service to our deployment GitHub Actions workflows:
+
+-   If it is only infrastructure it can be added to `./.github/workflows/deploy-infra-to-env.yml`.
+-   Services that include application code can be added to `./.github/workflows/deploy-app-to-env.yml`.
+-   We have a CI script that skips branch redeploys when possible in `./scripts/get-changed-services/index.ts`. Make sure your service is added to that list.
+
```
Comment (haworku) [inline README.md:273]: 🏅 gold medal documentation thank ya.

File: scripts/get-changed-services/index.ts [hunk 4]
```diff
@@ -4,23 +4,12 @@ import * as core from '@actions/core'
...
-    'run-migrations',
-    'prisma-layer',
-    'infra-api',
-]
-
 const octokit = new Octokit()
 
 async function main() {
+    // get our service names from lerna
+    const listOfServices = await getAllServicesFromLerna()
```
Comment (haworku) [inline scripts/get-changed-services/index.ts:11]: 🙌

**Pull request merged**